unstructured_ingest/v2/processes/connectors/onedrive.py

```diff
@@ -116,25 +117,34 @@ def precheck(self) -> None:
             logger.error(f"failed to validate connection: {e}", exc_info=True)
             raise SourceConnectionError(f"failed to validate connection: {e}")
 
-    def list_objects(self, folder: DriveItem, recursive: bool) -> list["DriveItem"]:
+    def list_objects_sync(self, folder: DriveItem, recursive: bool) -> list["DriveItem"]:
         drive_items = folder.children.get().execute_query()
         files = [d for d in drive_items if d.is_file]
         if not recursive:
             return files
+
         folders = [d for d in drive_items if d.is_folder]
         for f in folders:
-            files.extend(self.list_objects(f, recursive))
+            # Use the sync method instead of the async one here
+            files.extend(self.list_objects_sync(f, recursive))
         return files
 
-    def get_root(self, client: "GraphClient") -> "DriveItem":
+    async def list_objects(self, folder: "DriveItem", recursive: bool) -> list["DriveItem"]:
```

Comment:
I think it could be replaced with a decorator like:
```python
def asyncify(func):
    async def inner(*args, **kwargs):
        return await asyncio.to_thread(func, *args, **kwargs)
    return inner

@asyncify
def list_objects(...)
```
I'm just not sure what linters will think about it ;)

Reply:
I'll try and see! Looks better :)

Reply:
If you do - please use `from functools import wraps` to make the decorator elegant :)

Reply:
I've looked into it, feels like the wrap have to be a bit complicated, still looking into it